registration/models.py

```diff
@@ -91,8 +91,9 @@ class BetaTestApplication(ValidateModelMixin, TimeStampedModel):
                 'The subdomain name must at least have 3 characters.',
             ),
             validators.RegexValidator(
-                r'^[\w.-]+$',
-                "Please include only letters, numbers, '_', '-' and '.'",
+                r'^[a-zA-Z0-9]([a-zA-Z0-9\-]+[a-zA-Z0-9])?$',
```

Comment:
We don't need to allow uppercase letters here, if we're just going to lowercase them later anyway.

Reply:
I suppose this is a UX thing: do we want to tell the user they have to lowercase their chars? (This also means the error message has to change).

Reply:
Yep, I think it's ok to tell the user to lowercase their characters.

We also don't want to have to deal with uniqueness checks that fail *after* we lowercase their domains.

Reply:
Perhaps we can accept upper case letters but force the domain name to lower case via JavaScript on blur?

Reply:
That would work but is extra work for not that much gain I think :P